Source/RESTyard.Client.Extensions/SystemNetHttp/HttpHypermediaResolver.cs

```diff
@@ -165,14 +165,25 @@ protected override async Task EnsureRequestIsSuccessfulAsync(HttpResponseMessage
             var (hasProblemDescription, problemDescription) = await this.TryReadProblemStringAsync(responseMessage);
             if (hasProblemDescription)
             {
+                if (problemDescription.Status is not null)
+                {
+                    var httpStatus = (HttpStatusCode)problemDescription.Status;
+                    switch (httpStatus)
```

Comment:
we have many more specific exceptions, that are never used. The others can not be precisely mapped to one status code, or the other way around. Should we try to include more information at this level (e.g. adding an extension to ProblemDetails) to distinguish, or rather remove the not-mappable exceptions and let the consumer define what they need to figure out more about why the request did not succeed?